x-pack/plugins/observability/public/pages/slo_edit/components/apm_availability/apm_availability_indicator_type_form.tsx

```diff
@@ -5,21 +5,20 @@
  * 2.0.
  */
 
-import React, { useEffect } from 'react';
 import { EuiFlexGroup, EuiFlexItem, EuiIconTip } from '@elastic/eui';
-import { useFormContext } from 'react-hook-form';
 import { i18n } from '@kbn/i18n';
-import type { CreateSLOInput } from '@kbn/slo-schema';
-
+import React, { useEffect } from 'react';
+import { useFormContext } from 'react-hook-form';
 import { useFetchApmIndex } from '../../../../hooks/slo/use_fetch_apm_indices';
+import { CreateSLOForm } from '../../types';
 import { FieldSelector } from '../apm_common/field_selector';
 import { QueryBuilder } from '../common/query_builder';
 
 export function ApmAvailabilityIndicatorTypeForm() {
-  const { setValue, watch } = useFormContext<CreateSLOInput>();
+  const { setValue, watch } = useFormContext<CreateSLOForm>();
   const { data: apmIndex } = useFetchApmIndex();
   useEffect(() => {
-    setValue('indicator.params.index', apmIndex);
+    setTimeout(() => setValue('indicator.params.index', apmIndex));
```

Comment:
Good question, this one is tricky. When we first switch to an APM indicator, we have a RQ call to an internal endpoint to return the indices used by APM, that we store in a hidden input through the `setValue`.
But when we switch to another APM indicator, we reset the form so the previously stored `indicator.params.index` is wiped out but at the same time (or actually before), RQ returns the call with the indices instantly because it is cached, and it still get wiped.
With this, I'm ensuring we wipe the form, and then set the value of the index. Not ideal, but I don't have a better solution yet.
I'm gonna try again a few things 🤷🏻 because it bothers me

Reply:
@kdelemme, thank you for explaining that!  Yeah, `setTimeout ` is scary and could be unpredictable as it relays on the call stack.

I wonder how expensive it is to fetch APM indices. If we disable the RQ cache for that HTTP call, we no longer need that hidden input nor the `setTimeout`, Am I right?

Reply:
Maybe I'll try that

Reply:
@kdelemme, interesting! Can you give `refetchOnWindowFocus: true` a try?